File: src/main/java/com/nndmove/app/service/dto/PaymentDTO.java
package com.nndmove.app.service.dto;

import jakarta.validation.constraints.*;
import java.io.Serializable;
import java.util.Objects;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * A DTO for the {@link com.nndmove.app.domain.Payment} entity.
 */
@SuppressWarnings("common-java:DuplicatedBlocks")
@Getter
@Setter
@NoArgsConstructor
public class PaymentDTO implements Serializable {

    private Long id;

    @NotNull
    private Long paymentTime;

    @NotNull
    private Long paymentPrice;

    @NotNull
    private String paymentMethod;

    @NotNull
    private String status;

    @NotNull
    private String transactionId;

    private UserDTO user;

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof PaymentDTO)) {
            return false;
        }

        PaymentDTO paymentDTO = (PaymentDTO) o;
        if (this.id == null) {
            return false;
        }
        return Objects.equals(this.id, paymentDTO.id);
    }

    @Override
    public int hashCode() {
        return Objects.hash(this.id);
    }

    // prettier-ignore
    @Override
    public String toString() {
        return "PaymentDTO{" +
            "id=" + getId() +
            ", paymentTime=" + getPaymentTime() +
            ", paymentPrice=" + getPaymentPrice() +
            ", paymentMethod='" + getPaymentMethod() + "'" +
            ", status='" + getStatus() + "'" +
            ", transactionId='" + getTransactionId() + "'" +
            ", user=" + getUser() +
            "}";
    }
}

File: src/main/java/com/nndmove/app/service/dto/PaymentDTO.kt
package com.nndmove.app.service.dto

import jakarta.validation.constraints.NotNull
import java.io.Serializable
import java.util.*

/**
 * A DTO for the [com.nndmove.app.domain.Payment] entity.
 */
class PaymentDTO : Serializable {
    @JvmField
    var id: Long? = null

    @JvmField
    var paymentTime: @NotNull Long? = null

    @JvmField
    var paymentPrice: @NotNull Long? = null

    @JvmField
    var paymentMethod: @NotNull String? = null

    @JvmField
    var status: @NotNull String? = null

    @JvmField
    var transactionId: @NotNull String? = null

    @JvmField
    var user: UserDTO? = null

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o !is PaymentDTO) {
            return false
        }

        if (this.id == null) {
            return false
        }
        return this.id == o.id
    }

    override fun hashCode(): Int {
        return Objects.hash(this.id)
    }

    // prettier-ignore
    override fun toString(): String {
        return "PaymentDTO{" +
                "id=" + id +
                ", paymentTime=" + paymentTime +
                ", paymentPrice=" + paymentPrice +
                ", paymentMethod='" + paymentMethod + "'" +
                ", status='" + status + "'" +
                ", transactionId='" + transactionId + "'" +
                ", user=" + user +
                "}"
    }
}
